src/App.jsx

```diff
@@ -14,45 +18,102 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button
-        type="button"
-        className="button is-info is-light"
-      >
-        Sort alphabetically
-      </button>
-
-      <button
-        type="button"
-        className="button is-success is-light"
-      >
-        Sort by length
-      </button>
-
-      <button
-        type="button"
-        className="button is-warning is-light"
-      >
-        Reverse
-      </button>
-
-      <button
-        type="button"
-        className="button is-danger is-light"
-      >
-        Reset
-      </button>
-    </div>
+const SORT_BY_ALPHABET = 'alphabet';
+const SORT_BY_LENGTH = 'length';
+const SORT_BY_REVERSE_A = 'reverseA';
+const SORT_BY_REVERSE_B = 'reverseB';
+
+let sortedGoods = [...goodsFromServer];
+
+const getPreperedData = (goods, sortedBy, set) => {
+  const preperedGoods = [...goodsFromServer];
+
+  if (sortedBy === SORT_BY_REVERSE_A || sortedBy === SORT_BY_REVERSE_B) {
+    return sortedGoods.reverse();
+  }
+
+  if (sortedBy !== '') {
+    sortedGoods = preperedGoods.sort((good1, good2) => {
+      switch (sortedBy) {
+        case SORT_BY_ALPHABET:
+          return good1.localeCompare(good2);
+
+        case SORT_BY_LENGTH:
+          return good1.length - good2.length;
+
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (sortedBy === '') {
+    sortedGoods = [...goodsFromServer];
+
+    return sortedGoods;
+  }
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+  return preperedGoods;
+};
+
+export const App = () => {
+  const [sortedBy, setSortedBy] = useState('');
+  const goods = getPreperedData(goodsFromServer, sortedBy);
+
+  const reverse = () => {
+    if (sortedBy === SORT_BY_REVERSE_A) {
+      setSortedBy(SORT_BY_REVERSE_B);
+    } else {
+      setSortedBy(SORT_BY_REVERSE_A);
+    }
+  };
```

Comment:
Looks too complicated, create a boolean state for reverse and set it to true or false on button click